nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/service/ColdStorageServiceImpl.java

```diff
@@ -97,20 +104,88 @@
             VideoChangedListener.DISABLE_VIDEO_CONVERSIONS_GENERATION_LISTENER,
             PictureViewsGenerationListener.DISABLE_PICTURE_VIEWS_GENERATION_LISTENER);
 
+    /** @since 10.10 **/
+    public static final String COLDSTORAGE_RENDITION_EP = "coldStorageRendition";
+
+    /** @since 10.10 **/
+    protected static String defaultRendition;
+
+    /** @since 10.10 **/
+    protected static Map<String, String> renditionByDocType = new HashMap<>();
+
+    /** @since 10.10 **/
+    protected static Map<String, String> renditionByFacets = new HashMap<>();
+
     public ColdStorageServiceImpl() {
         // no instance allowed
     }
 
+    @Override
+    public void start(ComponentContext context) {
+        List< ColdStorageRenditionDescriptor> descriptors = getDescriptors(COLDSTORAGE_RENDITION_EP);
+        descriptors.forEach(descriptor -> {
+            String docType = descriptor.getDocType();
+            if (docType != null) {
+                renditionByDocType.put(docType, descriptor.getRenditionName());
+            }
+            String facet = descriptor.getFacet();
+            if (facet != null) {
+                renditionByFacets.put(facet, descriptor.getRenditionName());
+            }
+            if (docType == null && facet == null) {
+                defaultRendition = descriptor.getRenditionName();
+            }
+        });
+    }
+
+    public String getRenditionName(DocumentModel doc) {
+        if (renditionByDocType.containsKey(doc.getType())) {
+            return renditionByDocType.get(doc.getType());
+        }
+        for (Map.Entry<String, String> entry : renditionByFacets.entrySet()) {
+            if (doc.hasFacet(entry.getKey())) {
+                return entry.getValue();
+            }
+        }
+
+        if (defaultRendition == null) {
+            throw new NuxeoException("Please contribute a default rendition name");
+        }
+        return defaultRendition;
+    }
+
+    @Override
+    public Blob getRendition(DocumentModel doc, CoreSession session) {
+        // retrieve the thumbnail which will be used to replace the content, once the move done
+        Blob blob = Framework.getService(ThumbnailService.class).getThumbnail(doc, session);
```

Comment:
This is useless, see below: `renditionName` will never be null, so at worst, the default rendition (thumbnail if contributed) will be used.